crates/pathfinder/src/core.rs

```diff
@@ -153,3 +153,21 @@ pub struct EthereumTransactionIndex(pub u64);
 /// An Ethereum log's index within a block.
 #[derive(Debug, Copy, Clone, PartialEq, Hash, Eq)]
 pub struct EthereumLogIndex(pub u64);
+
+impl StarknetBlockNumber {
+    pub const GENESIS: StarknetBlockNumber = StarknetBlockNumber(0);
+
+    pub fn increment(&self) -> Self {
+        Self(self.0 + 1)
+    }
+
+    pub fn decrement(&self) -> Self {
+        Self(self.0 - 1)
+    }
```

Comment:
Earlier was about to suggest we just have 

```
impl Add<usize> for StarknetBlockNumber {
    type Output = StarknetBlockNumber;
    
    ...
}
```

And I guess we need the decrement here as well. It would be a nice place to make sure these are checked_add and checked_sub.